File: frontend/src/lib/api/promo-media.ts
import { apiFetch } from '@/lib/api/client'

export type PromoMediaRead = {
  id: string
  media_type: 'image' | 'video'
  url: string
  poster_url: string | null
  alt: string
  caption: string | null
  sort_order: number
  is_active: boolean
  created_at: string
}

export type PromoMediaInput = {
  media_type: 'image' | 'video'
  url: string
  poster_url?: string | null
  alt: string
  caption?: string | null
  sort_order?: number
  is_active?: boolean
}

/** Public: active showcase items for the Home page carousel. */
export function getPromoMedia() {
  return apiFetch<PromoMediaRead[]>('/promo-media')
}

/** Admin: all items, including inactive. */
export function getAllPromoMedia() {
  return apiFetch<PromoMediaRead[]>('/admin/promo-media')
}

export function createPromoMedia(payload: PromoMediaInput) {
  return apiFetch<PromoMediaRead>('/admin/promo-media', {
    method: 'POST',
    body: payload,
  })
}

export function deletePromoMedia(id: string) {
  return apiFetch<void>(`/admin/promo-media/${id}`, { method: 'DELETE' })
}

File: backend/app/schemas/promo_media.py
import uuid
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, ConfigDict, Field


class PromoMediaBase(BaseModel):
    media_type: Literal["image", "video"] = "image"
    url: str = Field(min_length=1, max_length=1024)
    poster_url: Optional[str] = Field(default=None, max_length=1024)
    alt: str = Field(min_length=1, max_length=500)
    caption: Optional[str] = None
    sort_order: int = 0
    is_active: bool = True


class PromoMediaCreate(PromoMediaBase):
    pass


class PromoMediaUpdate(BaseModel):
    media_type: Optional[Literal["image", "video"]] = None
    url: Optional[str] = Field(default=None, min_length=1, max_length=1024)
    poster_url: Optional[str] = Field(default=None, max_length=1024)
    alt: Optional[str] = Field(default=None, min_length=1, max_length=500)
    caption: Optional[str] = None
    sort_order: Optional[int] = None
    is_active: Optional[bool] = None


class PromoMediaRead(PromoMediaBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    created_at: datetime
